frontend/packages/kubevirt-plugin/src/components/modals/scheduling-modals/eviction-strategy-modal/eviction-strategy-modal.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalComponentProps,
+} from '@console/internal/components/factory';
+import { ModalFooter, ModalFooterProps } from '../../modal/modal-footer';
+import { withHandlePromise, HandlePromiseProps } from '@console/internal/components/utils';
+import { Checkbox, TextVariants, Text } from '@patternfly/react-core';
+
+import { k8sPatch } from '@console/internal/module/k8s';
+import { getEvictionStrategyPatch } from '../../../../k8s/patches/vm/vm-scheduling-patches';
+import { getVMLikeModel } from '../../../../selectors/vm/vmlike';
+
+import { VMLikeEntityKind } from '../../../../types/vmLike';
+
+import '../shared/scheduling-modals.scss';
+
+const EvictionStrategyModal: React.FC<EvictionStrategyModalProps> = withHandlePromise(
```

Comment:
No need for two types, just write
`withHandlePromise<EvictionStrategyModalWithPromiseProps>`